cmd/kured/main.go

```diff
@@ -199,13 +205,19 @@ func newCommand(name string, arg ...string) *exec.Cmd {
 
 // buildHostCommand writes a new command to run in the host namespace
 // Rancher based need different pid
-func buildHostCommand(pid int, command []string) []string {
+func buildHostCommand(pid int, command []string, GOOS string) []string {
 
-	// From the container, we nsenter into the proper PID to run the hostCommand.
-	// For this, kured daemonset need to be configured with hostPID:true and privileged:true
-	cmd := []string{"/usr/bin/nsenter", fmt.Sprintf("-m/proc/%d/ns/mnt", pid), "--"}
-	cmd = append(cmd, command...)
-	return cmd
+	if GOOS == windows {
+		// On Windows kubed daemonset needs to run in a HostProcess container which
+		// run all processes in the hosts process namespace.
+		return command
+	} else {
```

Comment:
Shouldn't we start to build an interface instead, and make sure the linker/user input allows us to infer the type of command to run?